pkg/controller/import-controller.go

```diff
@@ -105,35 +138,36 @@ func (ic *ImportController) processPvcItem(pvc *v1.PersistentVolumeClaim) error
 	if exists && (previousPhase == string(v1.PodSucceeded)) {
 		needsSync = false
 	}
-	if pod == nil && needsSync {
-		podEnvVar, err := createImportEnvVar(ic.clientset, pvc)
-		if err != nil {
-			return err
-		}
 
-		// all checks passed, let's create the importer pod!
-		ic.expectPodCreate(pvcKey)
-		pod, err = CreateImporterPod(ic.clientset, ic.image, ic.verbose, ic.pullPolicy, podEnvVar, pvc)
-		if err != nil {
-			ic.observePodCreate(pvcKey)
-			return err
-		}
-		return nil
+	if pod == nil && needsSync {
+		return ic.createImporterPod(pvc, pvcKey)
 	}
 
 	// update pvc with importer pod name and optional cdi label
 	anno := map[string]string{}
 	if pod != nil {
+		scratchExitCode := false
+		if pod.Status.Phase != v1.PodPending && pod.Status.ContainerStatuses != nil && pod.Status.ContainerStatuses[0].LastTerminationState.Terminated != nil &&
+			pod.Status.ContainerStatuses[0].LastTerminationState.Terminated.ExitCode > 0 {
```

Comment:
Did you look into using a pod termination message?  See here:  https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/

This way, we could pass obviously pass more information back to the controller.  Like, maybe (in the future) the pod could figure out exactly how big the scratch space needs to be.  Some sort of json termination message would be cool.

Reply:
I did, but decided it was over engineering at this point in time. I purposely kept this part pretty simple so its easy to change out to something more complicated in the future.